### ISSUE TITLE
SARIMAX predict breaks when simple_differencing=True

### ISSUE THREAD

**Issue by jamesuber** [statsmodels/statsmodels#2545]
In the diagnostics branch that is awaiting merge, sarimax.predict breaks when simple_differencing=True. You get the size matching error like

ValueError: Invalid dimensions for time-varying state intercept vector. Requires shape (*,673), got (50, 648)

This seems to be related to not adjusting properly for the points dropped from the prior differencing. Which seems to affect other things too. For example, the new plot_diagnostics has a similar problem when using simple_differencing. Seems that it may be related to loglikelihood_burn not being set properly?

**Comment by josef-pkt** [comment]
@ChadFulton

**Comment by ChadFulton** [comment]
Thanks for letting us know about this. It was a bug introduced when we recently refactored the way simple differencing affected the various places the datasets are stored.

The problem here is that I didn't update `SARIMAX.data.endog` as well as `SARIMAX.endog`. The `TimeSeriesModel._get_predict_end` method used the `data.endog` array to get `nobs` from, so it was using the pre-simple-differencing `nobs` and that made `predict` think we were forecasting and needed additional `exog` data.

Fixed in #2546, along with a few other cleanup-type things.
